Interrogator.xUnit/NotConcurrentAttribute.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Interrogator.xUnit.Common;
+using Interrogator.xUnit.Utilities;
+
+namespace Interrogator.xUnit
+{
+	internal static class TypeExtensions
+	{
+		public static IOrderedEnumerable<Type> OrderTypes(this IEnumerable<Type> types)
+			=> types.OrderBy(type => type.FullName);
+	}
+
+	[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+	public class NotConcurrentAttribute : DependsOnAttribute
+	{
+		private static readonly Dictionary<Type, MethodInfo[]> _methodInfoDictionary = new Dictionary<Type, MethodInfo[]>();
+
+		public enum ConcurrencyScope
+		{
+			Class,
+			ClassHierarchy,
+			Namespace,
+			Assembly
+		}
+
+		public string GroupName { get; }
+		public ConcurrencyScope Scope { get; }
+
+		public NotConcurrentAttribute() : this("", ConcurrencyScope.Class) { }
+		public NotConcurrentAttribute(string groupName) : this(groupName, ConcurrencyScope.Class) { }
+
+		public NotConcurrentAttribute(ConcurrencyScope scope) : this("", scope) { }
+
+		public NotConcurrentAttribute(string groupName, ConcurrencyScope scope) : base(null)
+		{
+			GroupName = $"{groupName}_{scope}"; // Adding scope to group name ensures that different scoped attributes don't collide
```

Comment:
Can you disallow null and empty string for groupname when it is specified please? Otherwise [NotConcurrent("")] will match [NotConcurrent]